include/mapbox/geojsonvt/tile.hpp

```diff
@@ -94,43 +95,43 @@ class InternalTile {
         return true;
     }
 
-    void addFeature(const vt_point& point, const property_map& props) {
-        tile.features.push_back({ transform(point), props });
+    void addFeature(const vt_point& point, const property_map& props, const std::experimental::optional<identifier>& id) {
+        tile.features.push_back({ transform(point), props, id });
     }
 
-    void addFeature(const vt_line_string& line, const property_map& props) {
+    void addFeature(const vt_line_string& line, const property_map& props, const std::experimental::optional<identifier>& id) {
         const auto new_line = transform(line);
         if (!new_line.empty())
-            tile.features.push_back({ std::move(new_line), props });
+            tile.features.push_back({ std::move(new_line), props, id });
     }
 
-    void addFeature(const vt_polygon& polygon, const property_map& props) {
+    void addFeature(const vt_polygon& polygon, const property_map& props, const std::experimental::optional<identifier>& id) {
         const auto new_polygon = transform(polygon);
         if (!new_polygon.empty())
-            tile.features.push_back({ std::move(new_polygon), props });
+            tile.features.push_back({ std::move(new_polygon), props, id });
     }
 
-    void addFeature(const vt_geometry_collection& collection, const property_map& props) {
+    void addFeature(const vt_geometry_collection& collection, const property_map& props, const std::experimental::optional<identifier>& id) {
         for (const auto& geom : collection) {
             vt_geometry::visit(geom, [&](const auto& g) {
                 // `this->` is a workaround for https://gcc.gnu.org/bugzilla/show_bug.cgi?id=61636
-                this->addFeature(g, props);
+                this->addFeature(g, props, id);
             });
         }
     }
 
     template <class T>
-    void addFeature(const T& multi, const property_map& props) {
+    void addFeature(const T& multi, const property_map& props, const std::experimental::optional<identifier>& id) {
```

Comment:
Maybe alias `std::experimental::optional<identifier>` to `optional_identifier`, which is repeated a lot?